### PR TITLE
TS Actions

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers

### CODE REVIEW

File: shared/actions/teams.tsx [hunk 1]
```diff
@@ -1334,15 +1362,18 @@ const badgeAppForTeams = (state, action) => {
...
+    // TODO ts-migration remove any
+    const toLoad: I.Set<any> = newTeamRequestsSet.subtract(existingNewTeamRequestsSet)
     const loadingCalls = toLoad.map(teamname => TeamsGen.createGetDetails({teamname})).toArray()
     actions = actions.concat(loadingCalls)
   }
 
   // if the user wasn't on the teams tab, loads will be triggered by navigation around the app
   actions.push(
     TeamsGen.createSetNewTeamInfo({
+      // @ts-ignore a legit error. todo fix
```
Comment (MarcoPolo) [inline shared/actions/teams.tsx]: @cjb I think this is a legit error from your PR. You forgot to add this in the types

File: shared/actions/chat2/index.tsx [hunk 2]
```diff
@@ -662,12 +701,40 @@ const onChatSetConvRetention = (_, action, logger) => {
   })
 }
 
-const onChatSetConvSettings = (_, action, logger) => {
+const onChatSetConvSettings = (_, action: EngineGen.Chat1NotifyChatChatSetConvSettingsPayload, logger) => {
   const {conv, convID} = action.payload.params
   const conversationIDKey = Types.conversationIDToKey(convID)
-  const newRole = conv?.convSettings?.minWriterRoleInfo?.role
+  const newRole = // Auto generated from flowToTs. Please clean me!
```
Comment (chrisnojima) [inline shared/actions/chat2/index.tsx]: i think we need to fix this

File: shared/actions/chat2/index.tsx [hunk 3]
```diff
@@ -2631,15 +2783,15 @@ const unsentTextChanged = (state, action) => {
   })
 }
 
-const onGiphyResults = (state, action) => {
+const onGiphyResults = (state, action: EngineGen.Chat1ChatUiChatGiphySearchResultsPayload) => {
```
Comment (chrisnojima) [inline shared/actions/chat2/index.tsx]: do we really have to type these actions also?

File: shared/package.json [hunk 4]
```diff
@@ -23,7 +23,7 @@
     "build-dev": "yarn run _helper build-dev",
     "build-emoji": "yarn run _node common-adapters/markdown/generate-emoji-parser.js",
     "build-prod": "yarn run _helper build-prod",
-    "build-actions": "yarn run _node desktop/yarn-helper/action-creator-creator",
+    "build-actions": "yarn run _node --extensions \".tsx\" desktop/yarn-helper/action-creator-creator.tsx",
```
Comment (chrisnojima) [inline shared/package.json]: i don't think its necessarily required converting all these files over unless its worth it

File: shared/util/flow.tsx [hunk 5]
```diff
@@ -1 +1,2 @@
-export const ifFlowComplainsAboutThisFunctionYouHaventHandledAllCasesInASwitch = (e: never) => {}
+// @ts-ignore codemod-issue
+export const ifFlowComplainsAboutThisFunctionYouHaventHandledAllCasesInASwitch = (e: never): never => {}
```
Comment (chrisnojima) [inline shared/util/flow.tsx]: i think we should toss this file

File: shared/actions/chat2/index.tsx (hunk 2 above)
Comment (MarcoPolo) [reply shared/actions/chat2/index.tsx]: I changed it to   `const cannotWrite = conv && conv.convSettings && conv.convSettings.minWriterRoleInfo && conv.convSettings.minWriterRoleInfo.cannotWrite || null` which is technically different than the nullish coalescing, but fine for this I think.

File: shared/util/flow.tsx (hunk 5 above)
Comment (MarcoPolo) [reply shared/util/flow.tsx]: We still need something like this for a similar reason. I was going to rename it in another pr.

File: shared/package.json (hunk 4 above)
Comment (MarcoPolo) [reply shared/package.json]: It was easy so why not /shrug. but good to know for future cases

File: shared/actions/chat2/index.tsx (hunk 3 above)
Comment (MarcoPolo) [reply shared/actions/chat2/index.tsx]: Yes. One of the differences in TS is that it won't infer function parameter types based on usage. (except in anonymous inline closures).

example: https://www.typescriptlang.org/play/#src=const%20numbers%20%3D%20%5B1%2C%202%2C%203%2C%204%2C%205%5D%0A%0Anumbers.map(a%20%3D%3E%20a%20%2B%201)%0A%0Aconst%20add1Untyped%20%3D%20(a)%20%3D%3E%20a%20%2B%201%0Aconst%20add1Typed%20%3D%20(a%3A%20number)%20%3D%3E%20a%20%2B%201%0A%0A%0Anumbers.map(add1Untyped)%0Anumbers.map(add1Typed)

File: shared/actions/fs/index.tsx [hunk 6]
```diff
@@ -154,7 +161,7 @@ const loadTlfSyncConfig = (state, action) => {
     .catch(makeUnretriableErrorHandler(action, tlfPath))
 }
 
-const setTlfSyncConfig = (state, action) =>
+const setTlfSyncConfig = (state, action: FsGen.SetTlfSyncConfigPayload) =>
```
Comment (MarcoPolo) [inline shared/actions/fs/index.tsx]: Basically just added the type of actions here. @jakob223 @songgao

File: shared/actions/teams.tsx [hunk 7]
```diff
@@ -1039,6 +1057,7 @@ function* setPublicity(state, action) {
           {
             name: teamname,
             settings: {
+              // @ts-ignore codemod fix when constants is typed
```
Comment (cjb) [inline shared/actions/teams.tsx]: fwiw constants is typed now

File: shared/actions/teams.tsx [hunk 8]
```diff
@@ -968,7 +981,12 @@ function* createChannel(_, action, logger) {
     }
 
     // Dismiss the create channel dialog.
-    if (Router2Constants.getVisibleScreen()?.routeName === 'chatCreateChannel') {
+    if (
+      // Auto generated from flowToTs. Please clean me!
+      (Router2Constants.getVisibleScreen() === null || Router2Constants.getVisibleScreen() === undefined
+        ? undefined
+        : Router2Constants.getVisibleScreen().routeName) === 'chatCreateChannel'
```
Comment (cjb) [inline shared/actions/teams.tsx]: I don't think we should merge in unconverted code like this, could be really unsafe

**Pull request merged**